app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -57,66 +56,35 @@ public function setBoltUserId($observer)
     }
 
     /**
-     * Event handler called after a save event.
-     * Calls the complete authorize Bolt end point to confirm the order is valid.
-     * If the order has been changed between the creation on Bolt end and the save in Magento
-     * an order info message is recorded to inform the merchant and a bugsnag notification sent to Bolt.
+     * Event handler called after Bolt confirms order authorization
+     *
+     * event: bolt_boltpay_authorization_after
      *
-     * @param $observer
-     * @throws Exception
+     * @param Varien_Event_Observer $observer Observer event contains `quote`, `order`, and the bolt transaction `reference`
+     *
+     * @throws Mage_Core_Exception if the bolt transaction reference is an object instead of expected string
      */
-    public function verifyOrderContents($observer)
+    public function completeAuthorize($observer)
```

Comment:
why is this called via bolt_boltpay_authorization_after event? given this is mostly operation around order to me it seems you could have this in Order.php